src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -209,6 +195,29 @@ public String getMonitoringEndpoint(Region region) {
     }
 
     private List<List<Dimension>> getDimensions(MetricRule rule, AmazonCloudWatchClient client) {
+        if (
+                rule.awsDimensions != null &&
+                rule.awsDimensions.size() == 1 &&
+                rule.awsDimensionSelect != null &&
+                rule.awsDimensionSelect.containsKey(rule.awsDimensions.get(0))
```

Comment:
What if there's more dimensions_select than that one provided?

Reply:
You're not checking for that case though.

Reply:
I think I am checkin it here:
```java
rule.awsDimensions.size() == 1
```

Or am I missing something.

Reply:
You're not doing that for awsDimensionSelect

Reply:
Why should I?
If the list contains `awsDimensions` only one dimension and this dimension is in the `awsDimensionSelect` map, why would we care about the other dimensions in this map?

Even in the current behavior, if the `awsDimensionSelect` contains keys which are not in the `awsDimensions`, those keys are currently ignored (as the list metrics did not select them).

Maybe I am mistaken (as I do not have many examples of cloudwatch metrics with multiple dimensions to test this with) and then I do not mind adding this extra check.

Reply:
I'm not sure exactly what the behaviour of cloudwatch will be here, but better to be safe than sorry if the user is doing something odd.